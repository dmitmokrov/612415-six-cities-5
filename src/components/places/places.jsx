import React from 'react';
import PropTypes from 'prop-types';
import SortingList from '../sorting-list/sorting-list';
import CardsList from '../cards-list/cards-list';
import withSortingList from '../../hocs/with-sorting-list/with-sorting-list';

const Places = (props) => {
  const {offers, sortType, city, isSortingListOpen, onTogglerClick} = props;

  return (
    <section className="cities__places places">
      <h2 className="visually-hidden">Places</h2>
      <b className="places__found">{offers.length} places to stay in {city}</b>
      <form className="places__sorting" action="#" method="get">
        <span className="places__sorting-caption">Sort by</span>
        <span className="places__sorting-type" tabIndex="0" onClick={onTogglerClick}>
          {sortType}
          <svg className="places__sorting-arrow" width="7" height="4">
            <use xlinkHref="#icon-arrow-select"></use>
          </svg>
        </span>
        {
          isSortingListOpen && <SortingList/>
        }
      </form>

      <div className="cities__places-list places__list tabs__content">
        <CardsList offers={offers}/>
      </div>
    </section>
  );
};

Places.propTypes = {
  city: PropTypes.string.isRequired,
  offers: PropTypes.array.isRequired,
  sortType: PropTypes.string.isRequired,
  isSortingListOpen: PropTypes.bool.isRequired,
  onTogglerClick: PropTypes.func.isRequired
};

export {Places};
export default withSortingList(Places);
