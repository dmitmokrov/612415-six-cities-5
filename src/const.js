export const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

export const AppRoute = {
  HOME: `/`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
  OFFER: `/offer/:id`
};

export const mainCardOptions = {
  cardClassName: `cities__place-card`,
  isHiddenPremiumMark: false,
  imageWrapperClassName: `cities__image-wrapper`,
  imageParams: {
    width: `260`,
    height: `200`
  },
  bookmarkPrefix: `To`
};

export const favoriteCardOptions = {
  cardClassName: `favorites__card`,
  isHiddenPremiumMark: true,
  imageWrapperClassName: `favorites__image-wrapper`,
  imageParams: {
    width: `150`,
    height: `110`
  },
  bookmarkPrefix: `In`
};

export const nearCardOptions = {
  cardClassName: `near-places__card`,
  isHiddenPremiumMark: true,
  imageWrapperClassName: `near-places__image-wrapper`,
  imageParams: {
    width: `260`,
    height: `200`
  },
  bookmarkPrefix: `To`
};
