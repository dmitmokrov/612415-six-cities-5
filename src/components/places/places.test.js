import React from 'react';
import renderer from 'react-test-renderer';
import {Places} from './places';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {NameSpace} from '../../store/reducers/root-reducer';
import {cities, sortTypes, AuthorizationStatus} from '../../const';

const offers = [
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatar: `img/1.png`,
      id: 3,
      isPro: true,
      name: `Angelina`
    },
    id: 1,
    images: [`img/1.png`, `img/2.png`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`
  },
  {
    bedrooms: 2,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      },
      name: `London`
    },
    description: `Big description`,
    goods: [`Coffee machine`, `Dishwasher`],
    host: {
      avatar: `img/1.png`,
      id: 3,
      isPro: true,
      name: `Angelina`
    },
    id: 3,
    images: [`img/1.png`, `img/2.png`],
    isFavorite: true,
    isPremium: true,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    },
    maxAdults: 2,
    previewImage: `img/12.png`,
    price: 1420,
    rating: 4.4,
    title: `Big Hotel`,
    type: `home`
  }
];
const sortType = `Popular`;
const city = `Amsterdam`;
const noop = () => {};

const mockStore = configureStore();

describe(`Should Places render correctly`, () => {
  it(`With opened sorting`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        offers: [],
        offer: null,
        nearbyOffers: [],
        comments: []
      },
      [NameSpace.PROCESS]: {
        city: cities[0],
        sortType: sortTypes[0],
        activeCardId: null
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH
      }
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Places
                offers={offers}
                sortType={sortType}
                city={city}
                isSortingListOpen={true}
                onTogglerClick={noop}
              />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With closed sorting`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        offers: [],
        offer: null,
        nearbyOffers: [],
        comments: []
      },
      [NameSpace.PROCESS]: {
        city: cities[0],
        sortType: sortTypes[0],
        activeCardId: null
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH
      }
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Places
                offers={offers}
                sortType={sortType}
                city={city}
                isSortingListOpen={false}
                onTogglerClick={noop}
              />
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
