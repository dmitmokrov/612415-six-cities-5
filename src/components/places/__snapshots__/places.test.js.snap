// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Places render correctly With closed sorting 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    2
     places to stay in 
    Amsterdam
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="place-card cities__place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__image-wrapper cities__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/1.png"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button  button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "96%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="place-card cities__place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="place-card__image-wrapper cities__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/12.png"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              1420
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "88%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            Big Hotel
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          home
        </p>
      </div>
    </article>
  </div>
</section>
`;

exports[`Should Places render correctly With opened sorting 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    2
     places to stay in 
    Amsterdam
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom places__options--opened"
    >
      <li
        className="places__option places__option--active"
        onClick={[Function]}
        tabIndex="0"
      >
        Popular
      </li>
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex="0"
      >
        Price: low to high
      </li>
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex="0"
      >
        Price: high to low
      </li>
      <li
        className="places__option "
        onClick={[Function]}
        tabIndex="0"
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="place-card cities__place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__image-wrapper cities__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/1.png"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button  button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "96%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="place-card cities__place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="place-card__image-wrapper cities__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/12.png"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              1420
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "88%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            Big Hotel
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          home
        </p>
      </div>
    </article>
  </div>
</section>
`;
